### ISSUE TITLE
Poco C++ Format Weird Behavior G++/MSVC

### ISSUE THREAD

**Issue by TylerTheFox** [pocoproject/poco#2280]
#### Expected behavior
Poco::Format operating the same way on Linux/Windows.

aka 
unsigned long long num = 20;
const auto test = Poco::format("TEST TEST %Lu", num); 

should result in a string test which contains "TEST TEST 20". This works on Windows but does not function on Linux instead results in [ERRFMT]. 

#### Actual behavior
![gcc](https://user-images.githubusercontent.com/6909044/38520677-4a99ece8-3bf8-11e8-8718-49ba032536ed.png)
![win32](https://user-images.githubusercontent.com/6909044/38520681-4c77d778-3bf8-11e8-987b-af456cba74b8.png)

#### POCO version
1.9.0

#### Compiler and version
Visual Studio 2017 (MSVC 15)
G++-5

#### Operating system and version
Ubuntu 16.04.04
Windows 10 Professional

App build command:
`g++ src/main.cpp -pthread -I extern/poco-1.9.0/Foundation/include -L extern/poco-1.9.0/lib/linux -std=c++11 -o bin/tipbot -lPocoJSON -lPocoUtil -lPocoNet -lPocoFoundation`

Poco just built it 
```
./configure
make
```

Contents of extern/poco-1.9.0/lib/linux

> brandan@DESKTOP-45NIHSE:~/IntenseCoin-TipBot/extern/poco-1.9.0/lib/linux$ ls
> libPocoEncodings.a   libPocoFoundation.a   libPocoJSON.a   libPocoNet.a   libPocoUtil.a   libPocoXML.a
> libPocoEncodingsd.a  libPocoFoundationd.a  libPocoJSONd.a  libPocoNetd.a  libPocoUtild.a  libPocoXMLd.a
> 

This is dynamically linked and output compile location files

> brandan@DESKTOP-45NIHSE:~/IntenseCoin-TipBot/bin$ ls
> libPocoEncodingsd.so  libPocoFoundationd.so  libPocoJSONd.so  libPocoNetd.so  libPocoUtild.so  libPocoXMLd.so  Makefile
> libPocoEncodings.so   libPocoFoundation.so   libPocoJSON.so   libPocoNet.so   libPocoUtil.so   libPocoXML.so   tipbot

**Comment by aleks-f** [comment]
while you're there, can you step into and see where exactly things go wrong? it will help to speed up the fix, otherwise, it will be fixed when we get to it. 

Thanks :)

**Comment by Brandantl** [comment]
I'll try I didn't copy the source files to my VM. I'll do that and get back.

**Comment by Brandantl** [comment]
Ok so I found a fix for it but it has to do wit hthe str << AnyCast<UInt64>(*itVal++); in formatOne inside Format.cpp.

I changed Uint64 to unsigned long long and now it operates correctly. Maybe an issue with Uint64 on Windows vs Linux? 

Before:
![before](https://user-images.githubusercontent.com/6909044/38524976-69fe3316-3c05-11e8-821d-31be81ecbc6b.PNG)

After:
![after](https://user-images.githubusercontent.com/6909044/38524981-6e21e83e-3c05-11e8-95a2-ae0192d49c6d.PNG)
